src/modules/EmailRecoveryModule.sol

```diff
@@ -43,16 +44,28 @@ contract EmailRecoveryModule is ERC7579ExecutorBase, IEmailRecoveryModule {
      */
     mapping(address account => bool isAuthorized) internal authorized;
 
-    event RecoveryExecuted();
+    event RecoveryExecuted(address indexed account, address indexed validator);
 
     error InvalidSelector(bytes4 selector);
+    error InvalidManager();
     error InvalidOnInstallData();
     error InvalidValidator(address validator);
     error NotTrustedRecoveryManager();
     error RecoveryNotAuthorizedForAccount();
 
     constructor(address _emailRecoveryManager, address _validator, bytes4 _selector) {
-        if (_selector == IModule.onUninstall.selector || _selector == IModule.onInstall.selector) {
+        if (_emailRecoveryManager == address(0)) {
+            revert InvalidManager();
+        }
+        if (_validator == address(0)) {
+            revert InvalidValidator(_validator);
+        }
+        if (
+            _selector == IModule.onUninstall.selector || _selector == IModule.onInstall.selector
+                || _selector == IERC7579Account.execute.selector
+                || _selector == ISafe.setFallbackHandler.selector
```

Comment:
Could you tell me why we need to check FallbackHandler.selector?

Reply:
It was a recommendation from the auditors. A 7579 Safe can be used as a validator. And that contract can execute arbitrary logic on a safe including the function - `setFallbackHandler`. Adding this check stops that function being called directly from the Safe on recovery as it could cause some damage if it was called